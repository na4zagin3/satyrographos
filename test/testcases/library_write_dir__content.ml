module StdList = List

open Satyrographos_testlib
open TestLib

open Shexp_process

let env ~dest_dir:_ ~temp_dir =
  let open Shexp_process.Infix in
  let empty_dist = FilePath.concat temp_dir "empty_dist" in
  PrepareDist.simple empty_dist
  >> return empty_dist

let () =
  let verbose = false in
  let main dist ~dest_dir ~temp_dir:_ ~outf =
    let open Satyrographos in
    let dest_dir = FilePath.concat dest_dir "dest" in
    let l = Library.read_dir ~outf dist in
    let l = { l with files =
      Core.Map.add_exn l.files ~key:"packages/test.satyg" ~data:(`Content "let x = 1")
    } in
    Library.write_dir ~outf ~verbose ~symlink:true dest_dir l in
  eval (test_install env main)
