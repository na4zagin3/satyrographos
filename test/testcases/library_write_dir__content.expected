Installing packages
------------------------------------------------------------
------------------------------------------------------------
@@dest_dir@@
@@dest_dir@@/dest
@@dest_dir@@/dest/fonts
@@dest_dir@@/dest/fonts/Junicode.ttf
@@dest_dir@@/dest/hash
@@dest_dir@@/dest/hash/fonts.satysfi-hash
@@dest_dir@@/dest/metadata
@@dest_dir@@/dest/packages
@@dest_dir@@/dest/packages/List.satyg
@@dest_dir@@/dest/packages/test.satyg
@@dest_dir@@/dest/unidata
@@dest_dir@@/dest/unidata/UnicodeData.txt
------------------------------------------------------------
diff -Nr @@empty_dir@@/dest/fonts/Junicode.ttf @@dest_dir@@/dest/fonts/Junicode.ttf
0a1
> @@Junicode.ttf@@
diff -Nr @@empty_dir@@/dest/hash/fonts.satysfi-hash @@dest_dir@@/dest/hash/fonts.satysfi-hash
0a1
> {"Junicode":<"Single":{"src":"dist/fonts/Junicode.ttf"}>}
diff -Nr @@empty_dir@@/dest/metadata @@dest_dir@@/dest/metadata
0a1,2
> ((version 1) (libraryName "") (libraryVersion "") (compatibility ())
>  (dependencies ()))
diff -Nr @@empty_dir@@/dest/packages/List.satyg @@dest_dir@@/dest/packages/List.satyg
0a1,2
> @stage: persistent
> module List = struct end
diff -Nr @@empty_dir@@/dest/packages/test.satyg @@dest_dir@@/dest/packages/test.satyg
0a1
> let x = 1
diff -Nr @@empty_dir@@/dest/unidata/UnicodeData.txt @@dest_dir@@/dest/unidata/UnicodeData.txt
0a1,3
> 0000;<control>;Cc;0;BN;;;;;N;NULL;;;;
> 0001;<control>;Cc;0;BN;;;;;N;START OF HEADING;;;;
> 0002;<control>;Cc;0;BN;;;;;N;START OF TEXT;;;;
